clients/java/src/main/java/io/camunda/zeebe/client/impl/ZeebeObjectMapper.java

```diff
@@ -57,6 +59,18 @@ public <T> T fromJson(final String json, final Class<T> typeClass) {
     }
   }
 
+  @Override
+  public <T> T fromJson(final String json,final Type type){
+
+    final JavaType objJavaType = objectMapper.constructType(type);
+    try {
+      return objectMapper.readValue(json, objJavaType);
+    } catch (final IOException e) {
+      throw new InternalClientException(
+          String.format("Failed to deserialize json '%s' to class '%s'", json, objJavaType), e);
+    }
+  }
```

Comment:
❌ It would be great if you could add a unit test to test this method to the [ZeebeObjectMapperTest](https://github.com/camunda/zeebe/blob/main/clients/java/src/test/java/io/camunda/zeebe/client/impl/ZeebeObjectMapperTest.java).